core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-hint-modal.controller.ts

```diff
@@ -50,7 +50,7 @@ angular.module('oppia').controller('AddHintModalController', [
 
     $scope.tmpHint = '';
     $scope.addHintForm = {};
-    $scope.hintIndex = StateHintsService.displayed.length + 1;
+    $scope.hintIndex = StateHintsService.getActiveHintIndex();
```

Comment:
Why this change?

Reply:
Also, have you tested the functionality to make sure there are no regressions?

Reply:
These changes are wrong, thanks for asking about it. I've fixed it now.

Reply:
Why was this not caught in the unit tests itself?

Reply:
Also, in future, make sure to do a self review using the files changed tab to ensure everything is exactly as you wanted it.

Reply:
Thanks @marianazangrossi! Just to check, does your commit 748b288 catch this? ie. if you change the code to the incorrect one, does the unit test fail? I'll approve once you confirm that is the case.

Reply:
Sure @nithusha21. Yes @kevintab95, the commit is catching this. It was working before because I was mocking it poorly to return `4` hard-coded.

Reply:
OK thanks for confirming!